lib/liquid/partial_cache.rb

```diff
@@ -2,18 +2,22 @@
 
 module Liquid
   class PartialCache
-    def self.load(template_name, context:, parse_context:)
+    def self.load(template_name, context:, parse_context:, caller:)
       cached_partials = (context.registers[:cached_partials] ||= {})
-      cached = cached_partials[template_name]
+      cached = cached_partials["#{caller}:#{template_name}"]
       return cached if cached
 
       file_system = (context.registers[:file_system] ||= Liquid::Template.file_system)
-      source      = file_system.read_template_file(template_name)
-
       parse_context.partial = true
 
-      partial = Liquid::Template.parse(source, parse_context)
-      cached_partials[template_name] = partial
+      source = if file_system.respond_to?(:read_template_file_with_options)
```

Comment:
I chose this approach to avoid breaking backwards compatibility. I am open to other ideas but this seemed like the safest at the moment. 

Also, I went with a rather generic name `read_template_file_with_options` to allow to pass in more options in the future if needed. I am open to being more specific if someone feels like this is too early of a generalization.